### PR TITLE
[Consensus] Add `ParentView` to `flow.Header`

### PR DESCRIPTION
**Pull request by durkmurder**
https://github.com/dapperlabs/flow-go/issues/6411

### Context

This PR adds `ParentView` to `flow.Header`. As part of this PR I have updated our test helpers, util functions, tests and wherever we have converted `flow.Header` to `model.Block` and vica versa. Also updated `hotstuff.BlockProducer` to return `flow.Header` instead of `model.Proposal`.

### CODE REVIEW

File: engine/collection/compliance/engine.go [hunk 1]
```diff
@@ -443,17 +443,6 @@ func (e *Engine) BroadcastProposalWithDelay(header *flow.Header, delay time.Dura
...
-	parent, err := e.headers.ByBlockID(header.ParentID)
-	if err != nil {
-		return fmt.Errorf("could not retrieve proposal parent: %w", err)
-	}
-
-	// fill in the fields that can't be populated by HotStuff
-	// TODO clean this up - currently we set these fields in builder, then lose them in HotStuff, then need to set them again here
-	header.ChainID = parent.ChainID
-	header.Height = parent.Height + 1
-
```
Comment (jordanschalm) [inline engine/collection/compliance/engine.go:456]: 🎉 Yay, I'm glad this is finally gone

File: go.mod [hunk 2]
```diff
@@ -56,12 +56,12 @@ require (
 	github.com/multiformats/go-multihash v0.2.1
 	github.com/onflow/atree v0.4.0
 	github.com/onflow/cadence v0.27.0
-	github.com/onflow/flow v0.3.2
+	github.com/onflow/flow v0.3.3-0.20221012172052-e4ab66320562
```
Comment (jordanschalm) [inline go.mod:59]: It would be nice to tag a version to pin here, before merging

File: go.mod [hunk 3]
```diff
@@ -56,12 +56,12 @@ require (
...
 	github.com/onflow/flow-core-contracts/lib/go/contracts v0.11.2-0.20220720151516-797b149ceaaa
 	github.com/onflow/flow-core-contracts/lib/go/templates v0.11.2-0.20220720151516-797b149ceaaa
 	github.com/onflow/flow-go-sdk v0.28.0
 	github.com/onflow/flow-go/crypto v0.24.4
-	github.com/onflow/flow/protobuf/go/flow v0.3.2-0.20220718115540-bd33a4f1cc17
+	github.com/onflow/flow/protobuf/go/flow v0.3.2-0.20221012172052-e4ab66320562
```
Comment (jordanschalm) [inline go.mod:64]: It would be nice to tag a version to pin here, before merging

**Review by jordanschalm — COMMENTED** [review]
👏 Thanks for this refactor - looks great. Two suggestions:
* Tag a new version in the two updated dependencies, and pin that version in our `go.mod`
  * `github.com/onflow/flow`
  * `github.com/onflow/flow/protobuf/go/flow`
* Add a consistency check for `ParentView` [in `Extend`](https://github.com/onflow/flow-go/blob/master/state/protocol/badger/mutator.go#L201-L204) and a test case for inconsistent `ParentView`

File: go.mod (hunk 3 above)
Comment (durkmurder) [reply go.mod:64]: Can you elaborate on this?
My plan was next:
1. `feature/active-pacemaker` points to specific commit of `onflow/flow`
2. `feature/active-pacemaker` gets merged into master
3. I create a PR in `onflow/flow` to merge branch into `master` and bump version. 
4. Update `go.mod` in `onflow/flow-go` on master branch to point to new version(without targeting specific commit). 

Does that make sense? 
Let me know how do you want to use tag here, currently github.com/onflow/flow/protobuf/go/flow v0.3.2-0.20221012172052-e4ab66320562 targets a commit on specific branch, it's not master.

Comment (jordanschalm) [reply go.mod:64]: 🤦 Of course you're right, shouldn't do this until we're merging the feature branch. Sorry, disregard that suggestion.

**Comment by AlexHentschel** [comment]
quick request, would you mind clearing up a tiny bit of tech debt as part of this PR 🙏 🙇 👍 
* in [this code](https://github.com/onflow/flow-go/blob/62b57dcdea46f005f21b2ed37737df347c89db98/consensus/recovery/follower.go#L18), the variable is still named `finalizer` (legacy naming). I think a correct name would be `forks`

**Review by AlexHentschel — APPROVED** [review]
This is such a great PR. Nice cleansing for core logic. ✨

File: consensus/hotstuff/follower_loop.go [hunk 4]
```diff
@@ -37,9 +37,9 @@ func NewFollowerLoop(log zerolog.Logger, followerLogic FollowerLogic) (*Follower
 //
 // Block proposals must be submitted in order, i.e. a proposal's parent must
 // have been previously processed by the FollowerLoop.
-func (fl *FollowerLoop) SubmitProposal(proposalHeader *flow.Header, parentView uint64) {
+func (fl *FollowerLoop) SubmitProposal(proposalHeader *flow.Header) {
 	received := time.Now()
-	proposal := model.ProposalFromFlow(proposalHeader, parentView)
+	proposal := model.ProposalFromFlow(proposalHeader)
 
 	fl.proposals <- proposal
```
[the remark this answers is not in the document]
Comment (durkmurder) [reply consensus/hotstuff/follower_loop.go]: I have covered it in this PR: https://github.com/onflow/flow-go/pull/3372/commits/3f44a0b5576cc343acf151ba3db59657e25c06a9

**Review by jordanschalm — APPROVED** [review]
🎸

**Comment by durkmurder** [comment]
bors merge

**Pull request merged**